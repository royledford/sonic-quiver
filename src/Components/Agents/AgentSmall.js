import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './AgentSmall.css'

export default class AgentSmall extends Component {
  static propTypes = {
    location: PropTypes.string.isRequired,
    companyName: PropTypes.string.isRequired,
    agentName: PropTypes.string.isRequired,
  }

  handleHover = () => {
    this.props.onMouseEnter(this.props.id)
  }

  render() {
    const { location, companyName, agentName } = this.props
    return (
      <div className="agentsmall-wrap" onMouseEnter={this.handleHover}>
        <h4 className="agentsmall-location">{location}</h4>
        <hr className="agentsmall-rule" />
        <p className="agentsmall-company">
          {companyName} - {agentName}
        </p>
      </div>
    )
  }
}
