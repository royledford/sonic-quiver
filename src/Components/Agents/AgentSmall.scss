@import 'styles/vars.scss';

.agentsmall-wrap {
  position: relative;
  max-width: 636px;
  height: 80px;
  padding: 20px 20px 10px 20px;
  color: $white;
  direction: ltr;
  transition: all 250ms ease-out;
  // Stop the contents from moving when the border is added.
  border: 1px solid rgba(0, 0, 0, 0);
}

.agentsmall-wrap:hover {
  background-color: $worldwide-overlay;
  border: 1px solid $white;
  transition: all 450ms ease-in-out;
}

.agentsmall-country {
  letter-spacing: 3px;
  position: relative;
  padding: 0;
  margin: 0 0 4px 0;
  font-size: 16px;
}

.agentsmall-copy {
  position: relative;
  padding: 0;
  margin: 6px 0 0 0;
  font-size: 12px;
}

.agentsmall-rule {
  height: 1px;
  background-color: rgba(255, 255, 255, 0.3);
  border-width: 0;
}

.agentsmall-mobile {
  display: none;
}

@media only screen and (max-width: 750px) {
  .agentsmall-wrap {
    height: 250px;

    background-color: $worldwide-overlay;
    border: 1px solid $white;
    border-bottom: none;
  }

  .agentsmall-mobile {
    display: block;
  }

  .agentsmall-wrap:hover {
    background-color: $worldwide-overlay;
    border: 1px solid $white;
    border-bottom: none;
    transition: none;
  }
}
