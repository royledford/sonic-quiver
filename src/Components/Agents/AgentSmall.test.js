import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'

import AgentSmall from './AgentSmall'
import renderer from 'react-test-renderer'

it('AgentSmall renders', () => {
  const tree = renderer
    .create(
      <AgentSmall
        country="country name"
        companyName="company"
        agentName="agent name"
        address1="address 1"
        address2="address 1"
        address3="address 1"
        phone="phone"
        fax="fax"
        email="email"
        urlDisplay="url display"
        sonicUrl="sonic url"
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
