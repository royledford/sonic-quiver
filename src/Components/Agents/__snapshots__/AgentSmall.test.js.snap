// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentSmall renders 1`] = `
<div
  className="agentsmall-wrap"
  onMouseEnter={[Function]}
>
  <h4
    className="country"
  >
    country name
  </h4>
  <hr
    className="agentsmall-rule"
  />
  <p
    className="agentsmall-copy"
  >
    company
     - 
    agent name
  </p>
  <div
    className="agentsmall-mobile"
  >
    <p>
      <a
        href="mailto:email"
      >
        email
      </a>
    </p>
    <a
      href="sonic url"
      rel="noopener noreferrer"
      target="_blank"
    >
      url display
    </a>
    <div
      className="agentsmall-copy"
    >
      TEL: 
      phone
      <br />
      FAX: fax
    </div>
    <div
      className="agentsmall-copy"
    >
      address 1
      <br />
      address 1
      <br />
      address 1
      <br />
    </div>
  </div>
</div>
`;
